### DISCUSSION TITLE
sync API prop names: rename client to queryClient

### DISCUSSION THREAD

**Discussion by milahu** [TanStack/query#4365] [Ideas]
mostly its queryClient
persistQueryClientSubscribe: props.queryClient

sometimes its client
QueryClientProvider: props.client

ideally its always queryClient
so i can avoid translation steps like

```js
mergedProps.queryClient = mergedProps.client
```

**Comment by TkDodo** [comment]
thanks, I'll keep in mind for v5, but I would opt for `client` instead of `queryClient`. Most functions have the name `queryClient` in the name already, like `QueryClientProvider` or `persistQueryClientSubscribe`. But I'll think about it.
